src/auction.js

```diff
@@ -0,0 +1,433 @@
+import { uniques, timestamp } from './utils';
+import { getPriceBucketString } from './cpmBucketManager';
+import { nativeBidIsValid } from './native';
+import { store } from './videoCache';
+import { Renderer } from 'src/Renderer';
+import { config } from 'src/config';
+
+const utils = require('./utils');
+const adaptermanager = require('./adaptermanager');
+// To be injected as dependency once #1453 is merged
+const events = require('./events');
+const CONSTANTS = require('./constants.json');
+
+const AUCTION_STARTED = 'started';
+const AUCTION_IN_PROGRESS = 'inProgress';
+const AUCTION_COMPLETED = 'completed';
+
+// register event for bid adjustment
+events.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+  adjustBids(bid);
+});
+
+function Auction(customPriceBucket, granularity) {
+  var _id = utils.getUniqueIdentifierStr();
+  var _adUnits = [];
+  var _adUnitCodes = [];
+  var _targeting = [];
+  var _bidderRequests = [];
+  var _bidsReceived = [];
+  var _auctionStart;
+  var _auctionId = utils.getUniqueIdentifierStr();
+  var _auctionStatus;
+  var _callback;
+  var _timer;
+
+  this.setAdUnits = (adUnits) => _adUnits = adUnits;
+  this.setAdUnitCodes = (adUnitcodes) => _adUnitCodes = adUnitcodes;
+  this.setTargeting = (targeting) => _targeting = targeting;
+  this.setBidderRequests = (bidderRequests) => _bidderRequests = _bidderRequests.concat(bidderRequests);
+  this.setBidsReceived = (bidsReceived) => _bidsReceived = _bidsReceived.concat(bidsReceived);
+  this.setAuctionId = (auctionId) => _auctionId = auctionId;
+  this.setAuctionStart = (auctionStart) => _auctionStart = auctionStart;
+  this.setAuctionStatus = (auctionStatus) => _auctionStatus = auctionStatus;
+  this.setCallback = (callback) => _callback = callback;
+  this.setTimer = (timer) => _timer = timer;
+
+  this.getId = () => _id;
+  this.getAdUnits = () => _adUnits;
+  this.getAdUnitCodes = () => _adUnitCodes;
+  this.getTargeting = () => _targeting;
```

Comment:
I feel like this API is just too... public.  It's hard to tell what is used where and I think since everything is getting exported publicly on this auction object it seems to suffer from a lack of encapsulation i.e. a lot of the logic related to auctions still seems to be scattered throughout other places in prebid.

I wonder how much this API can be simplified?  It's especially confusing because the auction manager seems to be re-exporting most of the auctions in aggregate and then other pieces of Prebid.js are then expecting to operate on those re-exported aggregates, like the targeting and such.